x-pack/plugins/observability/public/pages/slo_edit/components/slo_edit_form.tsx

```diff
@@ -121,64 +120,39 @@ export function SloEditForm({ slo }: Props) {
     const values = getValues();
 
     if (isEditMode) {
-      try {
-        const processedValues = transformValuesToUpdateSLOInput(values);
+      const processedValues = transformValuesToUpdateSLOInput(values);
 
-        await updateSlo({ sloId: slo.id, slo: processedValues });
+      await updateSlo({ sloId: slo.id, slo: processedValues });
 
-        toasts.addSuccess(
-          i18n.translate('xpack.observability.slo.sloEdit.update.success', {
-            defaultMessage: 'Successfully updated {name}',
-            values: { name: getValues().name },
-          })
+      if (isCreateRuleCheckboxChecked) {
+        navigate(
+          basePath.prepend(
+            `${paths.observability.sloEdit(slo.id)}?${CREATE_RULE_SEARCH_PARAM}=true`
+          )
         );
-
-        if (isCreateRuleCheckboxChecked) {
-          navigateToUrl(
-            basePath.prepend(
-              `${paths.observability.sloEdit(slo.id)}?${CREATE_RULE_SEARCH_PARAM}=true`
-            )
-          );
-        } else {
-          navigateToUrl(basePath.prepend(paths.observability.slos));
-        }
-      } catch (error) {
-        toasts.addError(new Error(error), {
-          title: i18n.translate('xpack.observability.slo.sloEdit.creation.error', {
-            defaultMessage: 'Something went wrong',
-          }),
-        });
+      } else {
+        navigate(basePath.prepend(paths.observability.slos));
       }
     } else {
-      try {
-        const processedValues = transformValuesToCreateSLOInput(values);
+      const processedValues = transformValuesToCreateSLOInput(values);
 
-        const { id } = await createSlo({ slo: processedValues });
+      const { id } = await createSlo({ slo: processedValues });
 
-        toasts.addSuccess(
-          i18n.translate('xpack.observability.slo.sloEdit.creation.success', {
-            defaultMessage: 'Successfully created {name}',
-            values: { name: getValues().name },
-          })
+      if (isCreateRuleCheckboxChecked) {
+        navigate(
+          basePath.prepend(`${paths.observability.sloEdit(id)}?${CREATE_RULE_SEARCH_PARAM}=true`)
         );
-
-        if (isCreateRuleCheckboxChecked) {
-          navigateToUrl(
-            basePath.prepend(`${paths.observability.sloEdit(id)}?${CREATE_RULE_SEARCH_PARAM}=true`)
-          );
-        } else {
-          navigateToUrl(basePath.prepend(paths.observability.slos));
-        }
-      } catch (error) {
-        toasts.addError(new Error(error), {
-          title: i18n.translate('xpack.observability.slo.sloEdit.creation.error', {
-            defaultMessage: 'Something went wrong',
-          }),
-        });
+      } else {
+        navigate(basePath.prepend(paths.observability.slos));
       }
     }
   };
 
+  const navigate = useCallback(
```

Comment:
Other solution is probably a `useState` / `useEffect` combo, but hate what that does for readability

Reply:
We could definitely. But I must admit, I don't fully understand the cause of these errors yet. If we manage to find that out, we might change this pattern to something different or remove it. We will revisit this though as it's just weird.